server/src/main/java/org/apache/iotdb/db/mpp/sql/planner/DistributionPlanner.java

```diff
@@ -94,6 +109,36 @@ public PlanNode visitDeviceMerge(TimeJoinNode node, DistributionPlanContext cont
       return null;
     }
 
+    @Override
+    public PlanNode visitMetaMerge(MetaMergeNode node, DistributionPlanContext context) {
+      MetaMergeNode root = (MetaMergeNode) node.clone();
+      MetaScanNode seed = (MetaScanNode) node.getChildren().get(0);
+      TreeSet<RegionReplicaSet> schemaRegions =
+          new TreeSet<>(Comparator.comparingInt(region -> region.getId().getId()));
+      analysis
+          .getSchemaPartitionInfo()
+          .getSchemaPartition()
+          .forEach(
+              (storageGroup, deviceGroup) -> {
+                deviceGroup.forEach(
+                    (deviceGroupId, schemaRegionReplicaSet) ->
+                        schemaRegions.add(schemaRegionReplicaSet));
+              });
+      int count = schemaRegions.size();
+      schemaRegions.forEach(
+          region -> {
+            MetaScanNode metaScanNode = (MetaScanNode) seed.clone();
+            metaScanNode.setRegionReplicaSet(region);
+            if (count > 1) {
+              metaScanNode.setLimit(metaScanNode.getOffset() + metaScanNode.getLimit());
+              metaScanNode.setOffset(0);
```

Comment:
The `limit` and `offset` params may need to be set in MetaMergeNode.

Reply:
`MetaMergeNode` is already a child of `OffsetNode`